mlflow/utils/virtualenv.py

```diff
@@ -108,21 +114,26 @@ def _install_python(version):
     # pyenv-win doesn't support `--skip-existing` but its behavior is enabled by default
     # https://github.com/pyenv-win/pyenv-win/pull/314
     pyenv_install_options = ("--skip-existing",) if _IS_UNIX else ()
+    extra_env = {"PYENV_ROOT": pyenv_root} if pyenv_root else None
+    pyenv_bin_path = _DATABRICKS_PYENV_BIN_PATH if is_in_databricks_runtime() else "pyenv"
     _exec_cmd(
-        ["pyenv", "install", *pyenv_install_options, version],
+        [pyenv_bin_path, "install", *pyenv_install_options, version],
         capture_output=False,
         # Windows fails to find pyenv and throws `FileNotFoundError` without `shell=True`
         shell=not _IS_UNIX,
+        extra_env=extra_env
     )
 
     if _IS_UNIX:
-        pyenv_root = _exec_cmd(["pyenv", "root"], capture_output=True).stdout.strip()
+        if pyenv_root is None:
+            pyenv_root = _exec_cmd([pyenv_bin_path, "root"], capture_output=True).stdout.strip()
         path_to_bin = ("bin", "python")
     else:
-        # pyenv-win doesn't provide the `pyenv root` command
-        pyenv_root = os.getenv("PYENV_ROOT")
         if pyenv_root is None:
-            raise MlflowException("Environment variable 'PYENV_ROOT' must be set")
+            # pyenv-win doesn't provide the `pyenv root` command
+            pyenv_root = os.getenv("PYENV_ROOT")
+            if pyenv_root is None:
```

Comment:
Do we need the second `if pyenv_root is None:`?

Reply:
We need to throw, but the following is probably cleaner:

```
pyenv_root = pyenv_root or os.getenv("PYENV_ROOT")
if pyenv_root is None:
    raise MlflowException("Environment variable 'PYENV_ROOT' must be set")
```

Reply:
I missed `pyenv_root = os.getenv("PYENV_ROOT")`. Yeah, that's cleaner.